### ISSUE TITLE
torch_ort.configure prints warning `cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++`

### ISSUE THREAD

**Issue by garymm** [microsoft/onnxruntime#9763]
Ideally there would be no warnings printed. This is the output I see:

```
➜  python -m torch_ort.configure
running build
running build_ext
building 'torch_gpu_allocator' extension
Emitting ninja build file /data/anaconda/envs/garymiguel-ZCode-MOE/lib/python3.6/site-packages/onnxruntime/training/ortmodule/torch_cpp_extensions/build/temp.linux-x86_64-3.6/build.ninja...
Compiling objects...Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)[1/1] c++ -MMD -MF /data/anaconda/envs/garymiguel-ZCode-MOE/lib/python3.6/site-packages/onnxruntime/training/ortmodule/torch_cpp_extensions/build/temp.linux-x86_64-3.6/data/anaconda/envs/garymiguel-ZCode-MOE/lib/python3.6/site-packages/onnxruntime/training/ortmodule/torch_cpp_extensions/cuda/torch_gpu
_allocator/torch_gpu_allocator.o.d -pthread -B /data/anaconda/envs/garymiguel-ZCode-MOE/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/bert_ort/garymiguel/src/pytorch/torch/include -I/bert_ort/garymiguel/src/pytorch/torch/include/torch/csrc/ap
i/include -I/bert_ort/garymiguel/src/pytorch/torch/include/TH -I/bert_ort/garymiguel/src/pytorch/torch/include/THC -I/usr/local/cuda/include -I/data/anaconda/envs/garymiguel-ZCode-MOE/include/python3.6m -c -c /data/anaconda/envs/garymiguel-ZCode-MOE/lib/python3.6/site-packages/onnxruntime/training/ort
module/torch_cpp_extensions/cuda/torch_gpu_allocator/torch_gpu_allocator.cc -o /data/anaconda/envs/garymiguel-ZCode-MOE/lib/python3.6/site-packages/onnxruntime/training/ortmodule/torch_cpp_extensions/build/temp.linux-x86_64-3.6/data/anaconda/envs/garymiguel-ZCode-MOE/lib/python3.6/site-packages/onnxru
ntime/training/ortmodule/torch_cpp_extensions/cuda/torch_gpu_allocator/torch_gpu_allocator.o -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1013"' -DTORCH_EXTENSION_NAME=torch_gpu_allocator -D_GLIBCXX_USE_CXX11_ABI=1 -st
d=c++14
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
```

**Comment by thiagocrepaldi** [comment]
If the code cant be improved to handle the warnings, https://github.com/microsoft/onnxruntime/blob/master/orttraining/orttraining/python/training/ortmodule/torch_cpp_extensions/cuda/torch_gpu_allocator/setup.py#L29-L30 is probably the spot to pass in C/LD flags to suppress it

Refer to `https://pytorch.org/docs/stable/cpp_extension.html#:~:text=torch.utils.cpp_extension.CUDAExtension%28name%2C%20sources%2C%20%2Aargs%2C%20%2A%2Akwargs%29%20%5Bsource%5D%20Creates%20a%20setuptools.Extension,CUDA%20include%20path%2C%20library%20path%20and%20runtime%20library.` to learn how to specify compiler/linker flags to Torch C++ extensions

**Comment by wschin** [comment]
This turns out to be [a 15-year old bug](https://bugs.python.org/issue1222585) from Python. [Pytorch] (https://github.com/pytorch/pytorch/commit/3ada9da808a16ba712242793d3b09c3205f78f41)workarounds this using
```
import distutils.sysconfig
cfg_vars = distutils.sysconfig.get_config_vars()
for key, value in cfg_vars.items():
    if type(value) == str:
        cfg_vars[key] = value.replace("-Wstrict-prototypes", "")
```

**Comment by garymm** [comment]
Actually it seems this may be fixed in some version of Python:
https://bugs.python.org/issue5755

My environment is using Python 3.6.13. Could someone check against 3.7? If it's fixed in 3.7, we should not take any action, since 3.6 is end-of-life in ~1 month.

**Comment by wschin** [comment]
This bug still exits in 3.7.9 (my environment).
